### PR TITLE
Ensure all upstream containers are running the same container version…

### LINKED ISSUE #37: How to handle service upgrades?
I'm curious if Interlock does anything to support atomically swapping out versions of an application.  For instance, suppose you have 1 container running version `A` of your application.  If you start another container running version `A`, it should be added to the same backend pool, so HAProxy can balance between them.  

However, if you then deploy a container running version `B`, you likely don't want it added to the same backend pool as the version `A`s. Doing so would cause problems with the wrong versions of assets being requested (if a request for `/s/B/main.js` was received by a backend running version `A`, it would return 404, or worse, return version `A`, which may then be cached by the CDN), and behavior flapping back and forth from one request to another.

I think the thing to do would be to atomically remove the `A` backend pool, and replace it with a `B` backend pool, when a version `B` container is seen.

Does Interlock address this?  I took a quick look through the code, and didn't see anything, but it is quite possible I missed it.  If you think it is the right approach, I can try to take a stab at a PR implementing my idea.

### PR DESCRIPTION
**Pull request by pkaeding**
… (Fixes #37)

### CODE REVIEW

File: plugins/haproxy/haproxy.go [hunk 1]
```diff
@@ -467,15 +470,17 @@ func (p HaproxyPlugin) GenerateProxyConfig() (*ProxyConfig, error) {
 			}
 		}
 
-	 	container_name := cInfo.Name[1:]
+		container_name := cInfo.Name[1:]
 		up := &Upstream{
 			Addr:          addr,
-                        Container:     container_name,
+			Container:     container_name,
```
Comment (ehazlett) [inline plugins/haproxy/haproxy.go]: I don't think this is from this PR but just noting here -- this should be `containerName`.

File: plugins/haproxy/haproxy.go [hunk 2]
```diff
@@ -508,6 +514,31 @@ func (p HaproxyPlugin) GenerateProxyConfig() (*ProxyConfig, error) {
...
+	for _, u := range ups {
+		if u != nil {
+			if u.Image == imageToKeep {
+				ret = append(ret, u)
+			} else {
+				logMessage(log.InfoLevel, fmt.Sprintf("Encountered %s running older container version: %s (newest is %s)", u.Container, u.Image, imageToKeep))
+			}
+		}
+	}
+	return
```
Comment (ehazlett) [inline plugins/haproxy/haproxy.go]: Naked return

**Comment by pkaeding** [comment]
Thanks for the feedback!  I just pushed a change addressing the issues you mentioned.

**Comment by pkaeding** [comment]
Ahh, I just realized there were conflicts here, and fixed them up.  Sorry for the delay!

**Comment by ehazlett** [comment]
Sorry for the delay :(  Closing as there has been a large refactor (`ng` branch).  Please re-open a new PR against that.  Thanks!

**Pull request closed without merging**